src/Serializer/DoctrineDenormalizer.php

```diff
@@ -63,7 +63,11 @@ public function denormalize($data, $class, $format = null, ?array $context = nul
 
             return $data;
         }
-
+        if (\is_string($data) === true) {
+            $key = \array_key_first($this->getClassLookupKey($class));
```

Comment:
I'd throw if there was more than a single key returned here - if we have a compound key it shouldnt query for half of that key

Reply:
Hang on, previously it just returned the raw data, and now we're throwing an error - which is it?

Reply:
Didnt think of that, however if we're trying to use the denormalizer for a compound key'd entity using scalars will not work

Reply:
I'll change it to not look up scalers in a multi-key entity then

Reply:
scalers when the entity requires composite values no longer trigger the lookup